Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -154,82 +206,90 @@ private ContentTagTask(CentralRepository db, PropertyChangeEvent evt) {
-                final ContentTagAddedEvent tagAddedEvent = (ContentTagAddedEvent) event;
-                final ContentTag tagAdded = tagAddedEvent.getAddedTag();
-
-                if (TagsManager.getNotableTagDisplayNames().contains(tagAdded.getName().getDisplayName())) {
-                    if (tagAdded.getContent() instanceof AbstractFile) {
-                        af = (AbstractFile) tagAdded.getContent();
-                        knownStatus = TskData.FileKnown.BAD;
-                        comment = tagAdded.getComment();
-                    } else {
-                        LOGGER.log(Level.WARNING, "Error updating non-file object");
-                        return;
-                    }
-                } else {
-                    // The added tag isn't flagged as bad in central repo, so do nothing
-                    return;
-                }
-            } else { // CONTENT_TAG_DELETED
-                // For deleted tags, we want to set the file status to UNKNOWN if:
-                //   - The tag that was just removed is notable in central repo
-                //   - There are no remaining tags that are notable 
-                final ContentTagDeletedEvent tagDeletedEvent = (ContentTagDeletedEvent) event;
+                handleTagAdded((ContentTagAddedEvent) event);
+            } 
+            else { 
+                handleTagDeleted((ContentTagDeletedEvent) event);
+            }
+        }
+        
+        private void handleTagDeleted(ContentTagDeletedEvent tagDeletedEvent) {
+            // CONTENT_TAG_DELETED
+            // For deleted tags, we want to set the file status to UNKNOWN if:
+            //   - The tag that was just removed is notable in central repo
+            //   - There are no remaining tags that are notable 
+            //   - If there are other notable tags, the comment should be updated to most recent tag
+
+            if (!isNotableTagName(tagDeletedEvent.getDeletedTagInfo().getName())) {
+                // If the tag that got removed isn't on the list of central repo tags, do nothing
+                return;
+            }
+
+            try {
+                // Get the remaining tags on the content object
                 long contentID = tagDeletedEvent.getDeletedTagInfo().getContentID();
+                Content content = Case.getCurrentCaseThrows().getSleuthkitCase().getContentById(contentID);
+                TagsManager tagsManager = Case.getCurrentCaseThrows().getServices().getTagsManager();
+                List<ContentTag> tags = tagsManager.getContentTagsByContent(content);
 
-                String tagName = tagDeletedEvent.getDeletedTagInfo().getName().getDisplayName();
-                if (!TagsManager.getNotableTagDisplayNames().contains(tagName)) {
-                    // If the tag that got removed isn't on the list of central repo tags, do nothing
+                AbstractFile af = getAbstractFile(content);
```

Comment:
getAbstractFile hides the fact that we are casting. It would be clearer if we used our knowledge that only AbstractFiles can be tagged, despite appearances to the contrary, by calling getAbstractFileById instead of getContentById.